GUI/Types/Exporter/GltfModelExporter.cs

```diff
@@ -130,16 +136,102 @@ private Mesh CreateGltfMesh(string meshName, VMesh vmesh, ModelRoot model)
                     primitive.WithIndicesAccessor(PrimitiveType.TRIANGLES, indices);
 
                     // Add material
-                    //var primitive = mesh.CreatePrimitive()
-                    //    .WithVertexAccessor("POSITION", positions)
-                    //    .WithIndicesAccessor(PrimitiveType.TRIANGLES, indices);
-                    //.WithMaterial(material);
+                    var materialPath = drawCall.GetProperty<string>("m_material");
+
+                    var renderMaterial = context.MaterialLoader.GetMaterial(materialPath);
+
+                    var materialNameTrimmed = Path.GetFileNameWithoutExtension(materialPath);
+                    var bestMaterial = GenerateGLTFMaterialFromRenderMaterial(renderMaterial, model, context, materialNameTrimmed);
+                    primitive.WithMaterial(bestMaterial);
                 }
             }
 
             return mesh;
         }
 
+        private Material GenerateGLTFMaterialFromRenderMaterial(RenderMaterial renderMaterial, ModelRoot model, VrfGuiContext context, string materialName)
+        {
+            var material = model
+                    .CreateMaterial(materialName + "_material")
```

Comment:
Is this _material postfix really needed, isn't it obvious this is a material?

Reply:
I assumed names had to be unique, but you are right. I'll change this for all materials, textures, images, and samplers